features/step_definitions/steps.rb

```diff
@@ -370,6 +370,14 @@
   create(:text_component, main_part: main_part, report: Report.current)
 end
 
+Given(/^there is an unpublished text component with the following main part:$/) do |main_part|
+  @text_component = create(:text_component, main_part: main_part, report: Report.current, publication_status: :draft)
+end
+
+Given(/I change the text component's publication status to "([^"]*)"$/) do |status|
+  @text_component.update! publication_status: status
```

Comment:
I would expect that `I change sth.` refers to a change in the user interface and not a low-level-change in the database (I am not a hacker!). I assume, the user interface is not yet finished?